protocol/src/protocol/mod.rs

```diff
@@ -1116,59 +1114,63 @@ pub struct Conn {
 }
 
 lazy_static! {
-    static ref IPADDRESS_PATTERN: Regex = Regex::new(
-        format!(
-            "{}{}{}{}",
-            "^([01]?\\d\\d?|2[0-4]\\d|25[0-5])\\.",
-            "([01]?\\d\\d?|2[0-4]\\d|25[0-5])\\.",
-            "([01]?\\d\\d?|2[0-4]\\d|25[0-5])\\.",
-            "([01]?\\d\\d?|2[0-4]\\d|25[0-5])$"
-        )
-        .as_str()
-    )
-    .unwrap();
     static ref RESOLVER: Resolver =
         Resolver::new(ResolverConfig::default(), ResolverOpts::default()).unwrap();
 }
 
 impl Conn {
-    fn get_server_addresses(mut hostname: &str) -> Vec<String> {
+    fn get_server_addresses(hostname: &str) -> Vec<(String, u16)> {
         let mut addresses = vec![];
-        let parts = hostname.split(':').collect::<Vec<&str>>();
-        if parts.len() > 1 {
-            addresses.push(hostname.to_string());
-            hostname = parts[0];
-        }
+
         let records = RESOLVER.srv_lookup(format!("_minecraft._tcp.{}", hostname));
         if records.is_ok() {
             for record in records.unwrap() {
                 debug!("{}:{}", record.target(), record.port());
-                addresses.push(format!("{}:{}", record.target(), record.port()));
+                addresses.push((record.target().to_string(), record.port()));
             }
         }
-        addresses.push(format!("{}:25565", hostname));
+        addresses.push((hostname.to_string(), 25565));
         addresses
     }
 
     pub fn new(target: &str, protocol_version: i32) -> Result<Conn, Error> {
         CURRENT_PROTOCOL_VERSION.store(protocol_version, Ordering::Relaxed);
-
         let mut address = target.to_string();
-        if !IPADDRESS_PATTERN.is_match(target) {
-            debug!("{} has an no address! :(", address);
-            let result = Conn::get_server_addresses(target);
-            // TODO: Try all possible ips not just the first!
-            let next = result.iter().next().unwrap();
-            debug!("{}'s ip may be {}.", address, next);
-            address = next.to_string();
+        let mut port = 25565;
+
+        if target.contains(':') {
```

Comment:
You could split_once directly here and use a if let instead of first checking contains and then splitting

Reply:
I don't even know why, I did not think about that xD.